example/src/main/deploy/openshift/enmasse_deploy.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+
+# Copyright (c) 2017 Red Hat and others.
+#
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#    Red Hat - initial creation
+#    Bosch Software Innovations GmbH
+
+# Absolute path this script is in
+SCRIPTPATH="$(cd "$(dirname "$0")" && pwd -P)"
+CONFIG=$SCRIPTPATH/../../config
+CERTS=$CONFIG/hono-demo-certs-jar
+HONO_HOME=$SCRIPTPATH/../../../..
+
+echo DEPLOYING ECLIPSE HONO ON OPENSHIFT
+
+# creating the directory for Hono Server persistent volume
+if [ ! -d /tmp/hono ]; then
+    mkdir /tmp/hono
+    chmod 777 /tmp/hono
+else
+    echo /tmp/hono already exists!
+fi
+
+# creating Hono persistent volume (admin needed)
+oc login -u system:admin
+oc create -f $SCRIPTPATH/hono-pv.yml
+
+oc login -u developer
+
+# use hono project
+oc project hono
+
+# create addresses
+curl -X PUT -T addresses.json -H "content-type: application/json" http://$(oc get route restapi -o jsonpath='{.spec.host}')/v1/addresses/default
```

Comment:
This invocation requires that there is already an "address-controller" pod in 'ready' state, which might not be the case yet when running this script directly after the EnMasse deployment script.
I haven't tested this with Openshift, but maybe adding a while loop with a test for the "address-controller" being ready might be a good idea here.

Reply:
@calohmn thanks for the feedback. I was testing this by making sure enmasse is fully deployed before going further. Adding checks definitely makes sense. I'll add it

Reply:
@dejanb @calohmn Have a look at https://github.com/EnMasseProject/enmasse/blob/master/systemtests/scripts/wait_until_up.sh#L5 for the check we use in systemtests to wait for address controller availability.

Moreover, you should also check the GET v1/addresses/default   output for status.isReady = true which indicates that the messaging infrastructure itself is ready.